frontend/src/contexts/user-context.tsx

```diff
@@ -0,0 +1,37 @@
+"use client";
+
+import { createContext, useContext, ReactNode, useState } from "react";
+
+interface UserContextType {
+  user: User | null;
+  setUser: React.Dispatch<React.SetStateAction<User | null>>;
+}
+
+const initialUser: User | null = null;
+
+/* eslint-disable @typescript-eslint/no-empty-function */
+const UserContext = createContext<UserContextType>({
+  user: initialUser,
+  setUser: () => {} // Placeholder function
+});
+/* eslint-enable @typescript-eslint/no-empty-function */
+
+function UserProvider({ children }: { children: ReactNode }) {
+  const [user, setUser] = useState<User | null>(initialUser);
+
+  return (
+    <UserContext.Provider value={{ user, setUser }}>
+      {children}
+    </UserContext.Provider>
+  );
+}
+
+const UseUserContext = () => {
```

Comment:
I believe this naming is not valid? Or at least is a bad practice. As useUserContext is supposed to be a React hook, kindly change the name to useUserContext instead.

Reply:
![image](https://github.com/tryyang2001/CS3213-Frontend-Management-System/assets/64810838/8bfba2fa-e1d4-4112-9323-8d42e6b6af1b)
Initially i went with `useUserContext` as we follow calmelCase naming standard, however this it the build error I get when i named it that way?

Reply:
I'm not familiar with useContext hook so if you have any workaround for this please let me know

Reply:
Mark as resolved, please check the later commits